chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -106,9 +107,26 @@
     private volatile TMIConnectionState connectionState = TMIConnectionState.DISCONNECTED;
 
     /**
-     * Channel Cache
+     * Channel Cache Lock
      */
-    protected final Set<String> channelCache = ConcurrentHashMap.newKeySet();
+    private final ReentrantLock channelCacheLock = new ReentrantLock();
+
+    /**
+     * Current Channels
+     */
+    protected final Set<String> currentChannels = ConcurrentHashMap.newKeySet();
+
+    /**
+     * Cache: ChannelId to ChannelName
+     */
+    @Getter
```

Comment:
A bit dangerous to have this getter public (and not wrapped in `UnmodifiableMap`) - same goes for the second map

Reply:
wrapped both getters with a unmodifiableMap